File: java/src/test/java/problems101_200/TwoSum2InputArrayIsSortedTest.java
package problems101_200;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class TwoSum2InputArrayIsSortedTest extends SampledTest {
    @Override
    protected String getSampleFilename() {
        return "two-sum-ii-input-array-is-sorted.json";
    }

    @Test
    public void test() {
        for (TwoSum2InputArrayIsSorted solution : TwoSum2InputArrayIsSorted.values()) {
            for (TwoSum2InputArrayIsSortedSample sample
                : getSamples(TwoSum2InputArrayIsSortedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.twoSum(sample.input.numbers, sample.input.target))
                    .asList()
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .boxed()
                            .toList()
                    ).inOrder();
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/TwoSum2InputArrayIsSortedTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class TwoSum2InputArrayIsSortedTest : SampledTest() {
    override fun getSampleFilename(): String = "two-sum-ii-input-array-is-sorted.json"

    @Test
    fun test() =
        TwoSum2InputArrayIsSorted.entries.forEach { solution ->
            getSamples<Array<TwoSum2InputArrayIsSortedSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.twoSum(input.numbers, input.target))
                        .isEqualTo(output)
                }
        }
}
